### ISSUE TITLE
Range constraint on date fields is always invalid

### ISSUE THREAD

**Issue by soerenbernstein** [formapro/JsFormValidatorBundle#126]
I have a symfony form which includes a DateType for a birthday field of my entity. The entity has an annotated Range constraint:
```
/**
  * @Assert\Range(
  *     max = "today",
  *     min = "first day of this month - 14 years UTC",
  */
  var $birthday;
```
This will work in symfony, but during client side validation, any date entered is invalid. Am I doing something wrong, or is this not supported?

**Comment by 66Ton99** [comment]
Yes, I think it is does not supported. Max and Min must be specific date.

**Comment by soerenbernstein** [comment]
Even with a specific date, this will not work because the js range constraint has some isNaN() test, which will fail on any date string. 
I've fiddled around with the code a bit and hack something that will allow dates to pass correctly, but only if the input field is of type "date". There are more issues, especially there is no way for the moment to remember the original (as in Symfony FormType) type of the field, so there is no distinction between DateTime, Date and Time.

**Comment by 66Ton99** [comment]
Fixed in [Svaroh/JsFormValidatorBundle](https://github.com/Svaroh/JsFormValidatorBundle), the new home of this bundle, by [PR #13](https://github.com/Svaroh/JsFormValidatorBundle/pull/13).

`Range` ran `parseFloat()` over its bounds, which cut `2017-06-29` down to `2017`, and then reported every date as not a number.

The bounds of a date field are now resolved on the server while the form is rendered - including the relative expressions the PHP date parser accepts, such as `today` or `first day of this month - 14 years UTC`, which no browser can parse - exported as concrete date strings, and compared as dates by the JavaScript constraint. `DateType` exports `Y-m-d`, `DateTimeType` exports `Y-m-d H:i:s`, and values of a different precision stay comparable.

A relative bound is frozen at the moment the page is rendered; the server re-reads it on submit and has the last word.

Documentation: [`3_18.md`](https://github.com/Svaroh/JsFormValidatorBundle/blob/main/src/Resources/doc/3_18.md).

The fix is in the `svaroh/jsformvalidator-bundle` package on `main`. It is not backported to `fp/jsformvalidator-bundle`, which stays on its 1.x line, so upgrading to the new package is what picks it up.
